Dockerfile

```diff
@@ -66,29 +124,9 @@ ENV PATH=${SDKMAN_DIR}/candidates/gradle/current/bin:${PATH}
 RUN \
      java --version \
   && gradle --version \
-  && add-apt-repository ppa:deadsnakes/ppa \
-  && apt-get update \
-  && apt-get install -y \
-      python-pip \
-      python3-pip \
-      python3.5-dev \
-      python3.6-dev \
-      python3.7-dev \
-      python3.8-dev \
-      python3.9-dev python3.9-dist \
   && pip3 install tox sphinx tox-run-command \
-  && curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain stable --no-modify-path \
-  && curl -sSL https://get.haskellstack.org/ | sh \
-  && rm -rf /var/lib/apt/lists/*
-
-ARG KITWARE_KEY_URL=https://apt.kitware.com/keys/kitware-archive-latest.asc
-
-RUN \
-     wget -O - ${KITWARE_KEY_URL} 2>/dev/null | gpg --dearmor - | sudo tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null \
-  && add-apt-repository 'deb https://apt.kitware.com/ubuntu/ bionic main' \ 
-  && apt-get update \
-  && apt-get install -y \
-    cmake
+  && curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain stable --profile minimal --no-modify-path \
```

Comment:
Do we do anything to pin the version (as opposed to the channel) of the rust toolchain we use? It should be alright to compile older projects with newer compilers because of the edition guarantees, so I don't think this is critical.

Reply:
This project seems to be ok with using stable-vs-nightly and edition as the only rustc versioning